### PR TITLE
[Lake][Enhancement] Make publish version more robust

### PR DESCRIPTION
**Pull request by sduzh**
0. Auto retry on network errors
1. Catch rpc exceptions
2. Avoid retry too often on failures
3. Avoid send duplicate publish requests

## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 28 / 113 (24.78%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/util/KafkaUtil.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/transaction/LakeTableTxnStateListener.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/rpc/LakeServiceClient.java|0|31|00.00%|
|:large_blue_circle:|com/starrocks/transaction/PartitionCommitInfo.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/common/proc/CurrentQueryInfoProvider.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/transaction/PublishVersionDaemon.java|0|37|00.00%|
|:large_blue_circle:|com/starrocks/qe/ResultReceiver.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/rpc/BackendServiceClient.java|5|10|50.00%|
|:large_blue_circle:|com/starrocks/rpc/BrpcProxy.java|22|27|81.48%|

File: fe/fe-core/src/main/java/com/starrocks/transaction/PublishVersionDaemon.java [hunk 1]
```diff
@@ -234,16 +234,26 @@ boolean publishTable(Database db, TransactionState txnState, TableCommitInfo tab
                 LOG.info("Removed partition {} from transaction {}", partitionId, txnId);
                 continue;
             }
-            if (!publishPartition(txnState, table, partition, partitionCommitInfo)) {
+            long currentTime = System.currentTimeMillis();
+            long versionTime = partitionCommitInfo.getVersionTime();
+            if (versionTime > 0) {
+                continue;
+            }
+            if (versionTime < 0 && currentTime + RETRY_INTERVAL_MS < Math.abs(versionTime)) {
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/transaction/PublishVersionDaemon.java:242]:
```suggestion
            if (versionTime < 0 && Math.abs(versionTime) + RETRY_INTERVAL_MS < currentTime) {
```

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 28 / 113 (24.78%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/util/KafkaUtil.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/transaction/LakeTableTxnStateListener.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/rpc/LakeServiceClient.java|0|31|00.00%|
|:large_blue_circle:|com/starrocks/transaction/PartitionCommitInfo.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/common/proc/CurrentQueryInfoProvider.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/transaction/PublishVersionDaemon.java|0|37|00.00%|
|:large_blue_circle:|com/starrocks/qe/ResultReceiver.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/rpc/BackendServiceClient.java|5|10|50.00%|
|:large_blue_circle:|com/starrocks/rpc/BrpcProxy.java|22|27|81.48%|

File: fe/fe-core/src/main/java/com/starrocks/transaction/PublishVersionDaemon.java (hunk 1 above)
Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/transaction/PublishVersionDaemon.java]: Updated, Should be `versionTime < 0 && currentTime < Math.abs(versionTime) + RETRY_INTERVAL_MS)`

File: fe/fe-core/src/main/java/com/starrocks/transaction/PublishVersionDaemon.java [hunk 2]
```diff
@@ -298,11 +316,25 @@ boolean publishPartition(TransactionState txnState, LakeTable table, Partition p
                     LOG.warn("Fail to publish tablet {} on BE {}", response.failedTablets, backendList.get(i).getHost());
```
[the remark this answers is not in the document]
Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/transaction/PublishVersionDaemon.java:316]: Yes, you're right, I plan to optimize this in the future.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 28 / 113 (24.78%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/util/KafkaUtil.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/transaction/LakeTableTxnStateListener.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/rpc/LakeServiceClient.java|0|31|00.00%|
|:large_blue_circle:|com/starrocks/transaction/PartitionCommitInfo.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/common/proc/CurrentQueryInfoProvider.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/transaction/PublishVersionDaemon.java|0|37|00.00%|
|:large_blue_circle:|com/starrocks/qe/ResultReceiver.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/rpc/BackendServiceClient.java|5|10|50.00%|
|:large_blue_circle:|com/starrocks/rpc/BrpcProxy.java|22|27|81.48%|

**Review by wyb — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Pull request merged**